nova/virt/vmwareapi/vmops.py

```diff
@@ -2181,22 +2169,38 @@ def is_instance_in_resource_pool(self, instance):
             LOG.debug("Failed to find instance", instance=instance)
             return False
 
-    def _get_instance_props(self, vm_ref):
-        lst_properties = ["config.instanceUuid",
-                          "runtime.powerState",
-                          "summary.guest.toolsStatus",
-                          "summary.guest.toolsRunningStatus",
-                         ]
+    def _get_instance_props(self, instance, lst_properties=None,
+                            skip_update=False):
+        lst_properties = (lst_properties or
+            ["config.instanceUuid",
+             "runtime.powerState",
+             "summary.guest.toolsStatus",
+             "summary.guest.toolsRunningStatus"])
 
-        self.update_cached_instances()
+        if not skip_update:
+            self.update_cached_instances()
+        vm_ref = vm_util.get_vm_ref(self._session, instance)
```

Comment:
This is fast because of the cache, right? Wondering because the method got `vm_ref` before.

Reply:
Yes, it should be in the same ball-park of speed thanks to the cache. The main point here was to have the `VMOps` work with `instance` as much as possible instead of `vm_ref`.